src/ServiceMap.php

```
@@ -119,10 +119,62 @@
 
+    public function getComponentPropertyById(string $id, string $propertyName): mixed
+    {
+        $definition = $this->components[$id] ?? null;
+
+        if ($definition === null) {
+            return null;
+        }
+
+        if (is_object($definition)) {
+            if (property_exists($definition, $propertyName)) {
+                return $definition->$propertyName;
```

PHPStan (2.1.15)

Variable property access on object.


(property.dynamicName)